src/usr/share/gpgmailer/gpgmailer-daemon.py

```diff
@@ -437,23 +439,29 @@ def create_spool_directories(program_uid, program_gid):
         raise exception
 
     spool_dir = os.path.join(SYSTEM_SPOOL_DIR, PROGRAM_NAME)
-    mounted_as_tmpfs = check_if_mounted_as_tmpfs(spool_dir)
 
-    # If directory is not mounted as tmpfs and there is something in the directory, fail to
-    #   start.
-    if os.listdir(spool_dir) != [] and not mounted_as_tmpfs:
-        raise InitializationException('Program spool directory is not empty and not mounted '
-                                      'as a ramdisk. Startup failed.')
+    if config['use_tmpfs_spool']:
+        mounted_as_tmpfs = check_if_mounted_as_tmpfs(spool_dir)
 
-    # If the program spool directory is empty and not already mounted as tmpfs, mount it as
-    #   tmpfs.
-    if not mounted_as_tmpfs:
-        logger.info('Attempting to mount the program spool directory as a ramdisk.')
-        subprocess.call(['mount', '-t', 'tmpfs', '-o', 'size=25%', 'none', spool_dir])
+        # If directory is not mounted as tmpfs and there is something in the directory, fail
+        #   to start.
+        # TODO: Do we still want to make sure the directory is empty? (I'm thinking probably
+        #   not, although, nothing should be in there unless something goes wrong or the flag
+        #   has just been set to use tmpfs and something was just queued.)
+        if os.listdir(spool_dir) != [] and not mounted_as_tmpfs:
+            raise InitializationException(
```

Comment:
If a message is queued and the the ramdisk was not mounted, would it be reasonable to move them to the ramdisk instead of simply failing?

Reply:
We talked about this and we decided to do what our other RamDisk implementation does. Just warn the user if there is something in the directory and proceed. (And remember to remove the TODO:.)